hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetworkTopology.java

```diff
@@ -539,10 +541,12 @@ protected Node chooseRandom(final String scope, String excludedScope,
         netlock.readLock().unlock();
       }
     }
-    LOG.debug("Choosing random from {} available nodes on node {},"
-        + " scope={}, excludedScope={}, excludeNodes={}. numOfDatanodes={}.",
-        availableNodes, innerNode, scope, excludedScope, excludedNodes,
-        numOfDatanodes);
+    if (LOG.isDebugEnabled()) {
```

Comment:
What are we saving by doing this?
The + should be statically handled by the compiler.